catalog/app/components/JsonEditor/ButtonMenu.js

```diff
@@ -17,13 +17,61 @@ const useStyles = M.makeStyles((t) => ({
   note: {
     fontFamily: t.typography.monospace.fontFamily,
     fontSize: t.typography.caption.fontSize,
-    marginRight: t.spacing(0.5),
+    marginLeft: t.spacing(0.5),
   },
 }))
 
-function ButtonMenu({ className, hasMenu, note, onClick }, ref) {
+function CellMenuItem({ item, onClick }) {
+  const onClickInternal = React.useCallback(() => onClick(item), [item, onClick])
+
+  // TODO: use PreviewValue function from Preview component
+  return (
+    <M.MenuItem onClick={onClickInternal}>
+      <M.ListItemText primary={item.title.toString()} />
+    </M.MenuItem>
+  )
+}
+
+function CellMenu({ anchorEl, menu, onClose, onClick }) {
+  if (!menu.length) return null
+
+  return (
+    <M.Menu anchorEl={anchorEl} onClose={onClose} open>
+      {menu.map((subList) => (
+        <M.List
+          subheader={
+            subList.header && <M.ListSubheader>{subList.header}</M.ListSubheader>
+          }
+          key={subList.key}
+        >
+          {subList.options.map((item) => (
+            <CellMenuItem
+              key={`${item.action}_${item.title}`}
+              item={item}
+              onClick={onClick}
+            />
+          ))}
+        </M.List>
+      ))}
+    </M.Menu>
+  )
+}
+
+export default function ButtonMenu({
+  className,
+  menu,
+  menuOpened,
+  note,
+  onClick,
+  onMenuClose,
+  onMenuSelect,
+}) {
   const classes = useStyles()
 
+  const ref = React.useRef(null)
+
+  const hasMenu = React.useMemo(() => !!menu.length, [menu])
```

Comment:
not very important, but anyways, `useMemo` isnt required here, because the computation (`!!`) is simple and result is a primitive value